website/docs/api/commands/cli.composite.md

```diff
@@ -126,6 +127,9 @@ ARGUMENTS
   
 OPTIONS
   -c, --ceramic-url        Ceramic API URL
+  -k, --did-private-key    DID Private Key (you can generate a fresh private key using composedb did:generate-private-key)
+  -i, --index              Adds the composite to indexing on the ceramic node (default)
+  --no-index               Prevents the composite from being indexed on the ceramic node. Useful if composite has already been deployed and indexed
```

Comment:
wait, what does it even mean to deploy a composite without indexing the models in it?

Reply:
> Useful if composite has already been deployed and indexed

If the composite was already deployed and indexed, why would you be running `composite:deploy` again at all?

Reply:
Possibly. They could have a CD process that just attempts to apply updates.

Reply:
what does that mean?  What do you mean by "apply updates"? Models can't be updated, so it's either deployed and indexed or not deployed and not indexed.  I still don't understand what it means to deploy without indexing.

Reply:
I just am still struggling to understand what deploying without indexing actually gains you.  It seems to me to be mostly a no-op.  If it doesn't do anything, then we shouldn't have people doing it.  If our docs need to be updated that's fine.

If there's some reason this is actually a useful feature other than "because our docs tell you to do it", then please let me know.  Maybe there's something I'm missing that this actually does which is useful.  But as of right now it doesn't seem to me like something there would ever be a legitimate reason to use.

Reply:
I think that's a question for @PaulLeCam 

I'm not understanding why we have a deploy command.

Reply:
> I'm not understanding why we have a deploy command.

I think this is the fundamental problem here, there seems to be a misunderstanding of how composites work.

There are 2 main parts to ComposeDB: the development tools that help manage the composites lifecycle and the runtime that generates a GraphQL schema using a compiled composite definition to interact with a Ceramic node.

In order for the runtime to work, it needs:
- A compiled composite definition, generated by `composite:compile` when using the CLI for example
- A Ceramic node indexing the models described in the composite, configured by running `composite:deploy` when using the CLI

So deploying a composite, whether by using the CLI or devtools libs, is needed to configure any Ceramic node to be usable by the ComposeDB runtime with an app's expected data model.
Creating a composite also deploys it to the given Ceramic node with the assumption that when developers create composites, they want to interact with the associated data as well. We can add an option not to index at this point as this PR does with the expectation developers will deploy the composite later, but the default should still be that a created composite is directly usable IMO.

Deploying a composite without indexing doesn't make sense to me, as it means the Ceramic node won't be able to handle queries from the ComposeDB runtime for the given composite. Do we have a specific use-case for this?

Reply:
It's not a misunderstanding, it's a question about something you specifically reference

> So deploying a composite, whether by using the CLI or devtools libs, is needed to configure any Ceramic node to be usable by the ComposeDB runtime with an app's expected data model.
Creating a composite also deploys it to the given Ceramic node with the assumption that when developers create composites

As it is, create does the same functionality as deploy currently. This PR makes it so create doesn't index unless specified, so that deploy is not a no-op. However, if we deploy the composite once, we can't deploy it again without passing the `--no-index` flag. 

We could remove the deploy command, make the default behavior on create to index, and have a flag not to index for repeated updates to a composite. The question is, why should we retain the deploy command? It doesn't seem to have any functionality other than starting indexing, which create does anyways.

Reply:
I just answered these questions in my previous comment, I don't really see how to explain it differently?

In a way deploying a composite is a bit like running `npm install` or similar to make sure an environment has the dependencies it requires to function. It is a no-op if the environment already has all the dependencies, but otherwise it's a necessary requirement to have the setup work as expected.

Reply:
> We could remove the deploy command, make the default behavior on create to index, and have a flag not to index for repeated updates to a composite. The question is, why should we retain the deploy command? It doesn't seem to have any functionality other than starting indexing, which create does anyways.

Yeah, it does seem like *either* we should just have the create command also control deploying/indexing the models with flags, *or* just always have a separate deploy command.

Personally I like keeping the separation between create and deploy as I do think those are conceptually different things and it seems clearer to me to keep them separate.  I also think it might be necessary for some workflows, such as when using a mix of pre-existing and net new models.  In that case you would want to use `composedb composite:from-model` to create a set of composites from the existing Models on the network you want to use, as well as use `composedb composite:create` to create a composite with all your new Models.  Then you probably want to use `composedb composite:merge` to merge them together, and then potentially update the merged composite to add any custom indices or views you want, and then only at the very end do you run `composedb composite:deploy` to actually deploy your final unified composite to the Ceramic node.

So my vote would be to make `composite:create` *not* deploy by default, so an explicit deploy is necessary.

I still don't think there's any reason to add a flag to deploy to allow deploying without indexing, because in my mind deploying == indexing.  The whole point of deploying a composite is to tell your Ceramic node to index the contained models.